lib/clients/BaseClient.py

```diff
@@ -531,6 +531,48 @@ def decrypt_and_extract_tarball_of_directory(self, encrypted_tarball_name, direc
             return result
         return None
 
+    def create_tarball_of_directory(self, directory_to_tar, tarball_name):
+        """Create a tarball of a directory.
+
+        :param directory_to_tar: the path to the directory to be archived
+        :param tarball_name: the path where to store the resulting archive
+
+        :Example:
+            ::
+                tar cvzpf ACS-Demo.tar.gz -C ACS-Demo .
+                iaas_client.create_tarball_of_directory('/tmp/backup/files.tar.gz', '/var/vcap/store/blueprint/files')
+        """
+        self.logger.info(
+            '[COMPRESSION] Started creating and copying a tarball ...')
+        result = self.shell('tar cvzpf {} -C {} .'
+                            .format(tarball_name, directory_to_tar), False)
+        self.logger.info('[ENCRYPTION] ... finished.')
+        return result
+
+    def extract_tarball_of_directory(self, tarball_name, directory_to_extract):
+        """Extract an encrypted tarball of a directory.
+
+        :param arball_name: the path to the directory to be extracted
+        :param directory_to_extract: the path to a directory where to move the extracted files
+
+        :Example:
+            ::
+
```

Comment:
Can you please add an example like [this](https://github.com/cloudfoundry-incubator/service-fabrik-backup-restore/pull/19/files#diff-973b6bc10f6b36c84d0dbeb7f11dc11aR542) ?

Reply:
@sauravmndl in my opinion that is not required as it is what function is supposed to do and not something for caller to take care. See my above comment.